controllers/main.go

```diff
@@ -79,10 +80,13 @@ func randToken() string {
 // empty, http.Request.RemoteAddr. See the sample nginx.conf for using the
 // real_ip module to correctly set the X-Real-IP header.
 func getClientIP(r *http.Request) string {
-	getHost := func(ip string) string {
-		if strings.Contains(ip, ":") {
-			parts := strings.Split(ip, ":")
-			return parts[0]
+	// getHost returns the host portion of a string containing either a
+	// host:port formatted name or just a host. An empty string is returned if
+	// net.SplitHostPort returns an error.
+	getHost := func(hostPort string) string {
+		ip, _, err := net.SplitHostPort(hostPort)
+		if err != nil {
+			return ""
```

Comment:
If someone is proxied and they set it up right, there's no need to check anything as there's no possibility for the client to spoof the header - it's set by the proxy.  Sounds like we want to handle the case of someone who's not proxied so that they don't get a spoofed X-Real-IP.  Assuming we want to support running w/o a proxy, then all that's needed is a boolean flag.